src/components/FilterBar/FilterBar.tsx

```diff
@@ -0,0 +1,241 @@
+import {
+  Box,
+  chakra,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useMultiStyleConfig,
+  Wrap,
+  WrapItem,
+} from "@chakra-ui/react";
+import React, { forwardRef } from "react";
+
+import Button from "../Button/Button";
+import ButtonGroup from "../ButtonGroup/ButtonGroup";
+import Heading from "../Heading/Heading";
+import { LayoutTypes } from "../../helpers/types";
+import MultiSelect from "../MultiSelect/MultiSelect";
+import MultiSelectGroup from "../MultiSelectGroup/MultiSelectGroup";
+import { SelectedItems } from "../MultiSelect/MultiSelect";
+import useNYPLBreapoints from "../../hooks/useNYPLBreakpoints";
+
+interface FilterBarCommonProps {
+  children: React.ReactNode;
+  id?: string;
+  isOpen: boolean;
+  headingText: string;
+  layout?: LayoutTypes;
+  selectedItems: SelectedItems;
+}
+
+type ClearAllProps =
+  | {
+      showClearAll?: false;
+      onClear?: never;
+    }
+  | { showClearAll: true; onClear: () => void };
+
+type SubmitAllProps =
+  | {
+      showSubmitAll?: false;
+      onSubmit?: never;
+    }
+  | { showSubmitAll: true; onSubmit: () => void };
+
+type HandleModalProps =
+  | {
+      onToggle: () => void;
+      onOpen?: never;
+      onClose?: never;
+    }
+  | {
+      onToggle?: undefined;
+      onOpen: () => void;
+      onClose: () => void;
+    };
+
+export type FilterBarProps = FilterBarCommonProps &
+  ClearAllProps &
+  SubmitAllProps &
+  HandleModalProps;
+
+/**
+ * `FilterBar` is a wrapper component for filter components.
+ * The DS considers the following components as filter components:
+ * - `MultiSelectGroup`
+ *
+ * The wrapped components/ component groups can be displayed in a column or
+ * in a row layout. `FilterBar` can render additional `Clear All` and a `Apply Filters` buttons. The two
+ * _optional_ buttons are controlled by the `showClearAll`/ `onClear` or `showSubmitAll`/`onSubmit` props repectively.
+ */
+
+export const FilterBar = chakra(
+  forwardRef<HTMLDivElement, React.PropsWithChildren<FilterBarProps>>(
+    (props, ref?) => {
+      const {
+        children,
+        id,
+        isOpen,
+        headingText,
+        layout = "row",
+        onClear,
+        onSubmit,
+        onToggle,
+        onOpen,
+        onClose,
+        selectedItems,
+        showClearAll = false,
+        showSubmitAll = false,
+        ...rest
+      } = props;
+
+      const { isLargerThanMedium } = useNYPLBreapoints();
+      const finalLayout = isLargerThanMedium ? layout : "column";
+      const styles = useMultiStyleConfig("FilterBar", { layout: finalLayout });
+      const getSelectedItemsCount = () =>
+        Object.entries(selectedItems).length
+          ? `(${Object.entries(selectedItems).length})`
+          : "";
+      //@TODO: resolved using Typescript descriminated unions
```

Comment:
Try creating a "type guard". See the `isFilterType` function here: https://github.com/NYPL/nypl-design-system/blob/development/src/components/TagSet/TagSet.tsx#L71